topcoder-platform/tc-message-service
# add fixes for message services

verification Steps-
setup docker containers and discourse as per readme
add postman collection and environment to postman
in postman:
  try creating, listing, getting topics
  (in discourse you can see that new user user is created and to view topics you can impersonate user from admin and see private messages in sentbox)
  to reply(post) replace 35 if url to topicId of topic you want to post
  in discourse by impersonating user can verify that @handle has changed to @userId( as of now @test_user1 and @magrathean have same userId due to mock service)
  when you try to get a topic or post or list topic,all @userId mentions will be converted back to @handles

sso logic is switched from handle to userId and vice versa


One time script is located at root of project simply run `node handleToUserId.js` to map handle to userId in db and rename users in Discourse
(make sure you have createdBy and updatedBy columns with user handles)

Extra:
added test db (messages_test), so that data is not lost everytime we run test
added joi validation to Post(reply api)
added reply(post) and getpost api to postman
